### PR TITLE
Convert Enzyme UsersTable test to RTL

### LINKED ISSUE #48255: Convert `public/app/features/users/UsersTable.test.tsx` to RTL

### PR DESCRIPTION
**Pull request by polibb**
**What this PR does / why we need it**:

Converts Enzyme test of UsersTable to [RTL](https://github.com/testing-library/react-testing-library)

**Which issue(s) this PR fixes**:

Fixes [#48255](https://github.com/grafana/grafana/issues/48255)

**Special notes for your reviewer**:

### CODE REVIEW

**Review by tolzhabayev — APPROVED** [review]

File: public/app/features/users/UsersTable.test.tsx [hunk 1]
```diff
@@ -15,39 +15,45 @@ jest.mock('app/core/core', () => ({
...
-};
+  return Object.assign(props, propOverrides);
+}
 
 describe('Render', () => {
   it('should render component', () => {
-    const wrapper = setup();
-
-    expect(wrapper).toMatchSnapshot();
+    render(<UsersTable {...getProps()} />);
```
Comment (ashharrison90) [inline public/app/features/users/UsersTable.test.tsx:30]: this test doesn't have an assertion. try to make sure your tests always contain an `expect` 
```suggestion
    expect(() => render(<UsersTable {...getProps()} />)).not.toThrow();
```

File: public/app/features/users/UsersTable.test.tsx [hunk 2]
```diff
@@ -15,39 +15,45 @@ jest.mock('app/core/core', () => ({
...
+        {...getProps({
+          users: getMockUsers(3),
+        })}
+      />
+    );
+    const user = userEvent.setup();
+
+    await user.click(screen.getAllByRole('button', { name: /delete/i })[0]);
+
+    expect(screen.getAllByText(/sure/i).length).toEqual(1);
```
Comment (ashharrison90) [inline public/app/features/users/UsersTable.test.tsx:57]: minor suggestion:

```suggestion
    expect(screen.getByText(/are you sure you want to delete user/i)).toBeInTheDocument();
```

File: public/app/features/users/UsersTable.test.tsx [hunk 3]
```diff
@@ -15,39 +15,45 @@ jest.mock('app/core/core', () => ({
...
+    render(<UsersTable {...getProps()} />);
   });
 
   it('should render users table', () => {
-    const wrapper = setup({
-      users: getMockUsers(5),
-    });
-
-    expect(wrapper).toMatchSnapshot();
+    render(
```
Comment (ashharrison90) [inline public/app/features/users/UsersTable.test.tsx:34]: same here, no `expect`.

sounds like it wants to check that the users are rendered in the table, so maybe something like:

```typescript
const users = getMockUsers(5);
render(<UsersTable {...getProps({ users })} />);
users.forEach((user) => {
  expect(screen.getByText(user.name)).toBeInTheDocument();
])
```

**Review by ashharrison90 — CHANGES_REQUESTED** [review]
great start! couple of minor comments, otherwise looks good 🥳

File: public/app/features/users/UsersTable.test.tsx [hunk 4]
```diff
@@ -15,39 +15,45 @@ jest.mock('app/core/core', () => ({
   },
 }));
 
-const setup = (propOverrides?: object) => {
+function getProps(propOverrides?: object) {
```
Comment (Clarity-89) [inline public/app/features/users/UsersTable.test.tsx:18]: You could also use `setup` function and render the component here instead like it was done originally, so you don't need to write same code multiple times.

File: public/app/features/users/UsersTable.test.tsx (hunk 3 above)
Comment (polibb) [reply public/app/features/users/UsersTable.test.tsx:34]: Yes! Okay. I wasn't sure how much should I rely on the current implementation, e.g. the fact that each user's name is visible only ones in the DOM right now

**Review by ashharrison90 — APPROVED** [review]
looks great! 🙌

**Review by Clarity-89 — APPROVED** [review]
Looks good! You'd also update snapshots (`u` key after running `yarn test`) and push the changes to this branch.

**Pull request merged**